src/main/java/gov/usgs/earthquake/product/io/JsonProduct.java

```diff
@@ -0,0 +1,263 @@
+package gov.usgs.earthquake.product.io;
+
+import gov.usgs.earthquake.product.AbstractContent;
+import gov.usgs.earthquake.product.Content;
+import gov.usgs.earthquake.product.Product;
+import gov.usgs.earthquake.product.ProductId;
+import gov.usgs.earthquake.product.URLContent;
+import gov.usgs.util.CryptoUtils.Version;
+import gov.usgs.util.protocolhandlers.data.Handler;
+import gov.usgs.util.StreamUtils;
+import gov.usgs.util.XmlUtils;
+
+import java.math.BigDecimal;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonValue;
+
+public class JsonProduct {
+
+  static {
+    // make sure data protocol handler registered
+    Handler.register();
+  }
+
+  /**
+   * Convert product object to json.
+   *
+   * @param product
+   * @return
+   * @throws Exception
+   */
+  public JsonObject getJsonObject(final Product product) throws Exception {
+    JsonObjectBuilder json = Json.createObjectBuilder();
+
+    json.add("contents", getContentsJson(product.getContents()));
+    JsonObjectBuilder geometry = getGeometryJson(product);
+    if (geometry == null) {
+      json.addNull("geometry");
+    } else {
+      json.add("geometry", geometry);
+    }
+    final ProductId id = product.getId();
+    json.add("id", getIdJson(id));
+    json.add("links", getLinksJson(product.getLinks()));
+    json.add("properties", getPropertiesJson(product.getProperties()));
+    json.add("signature", product.getSignature());
+    json.add("signatureVersion", product.getSignatureVersion().toString());
+    json.add("status", product.getStatus());
+    json.add("type", "Feature");
+    return json.build();
+  }
+
+  /**
+   * Convert json object to product.
+   */
+  public Product getProduct(final JsonObject json) throws Exception {
+    Product product = new Product(getId(json.getJsonObject("id")));
+    product.setContents(getContents(json.getJsonArray("contents")));
+    product.setLinks(getLinks(json.getJsonArray("links")));
+    product.setProperties(getProperties(json.getJsonObject("properties")));
+    product.setStatus(json.getString("status"));
+    product.setSignature(json.getString("signature"));
+    product.setSignatureVersion(Version.fromString(json.getString("signatureVersion")));
+    product.setTrackerURL(new URL("data:,"));
+    return product;
+  }
+
+  /**
+   * Convert contents map to json.
+   *
+   * @param contents
+   * @return
+   * @throws Exception
+   */
+  public JsonArrayBuilder getContentsJson(final Map<String, Content> contents) throws Exception {
+    final JsonArrayBuilder builder = Json.createArrayBuilder();
+    for (final String path : contents.keySet()) {
```

Comment:
Looping over the `entrySet` would be better since the code needs both keys and values.